faust/transport/consumer.py

```diff
@@ -749,10 +749,17 @@ async def getmany(self, timeout: float) -> AsyncIterator[Tuple[TP, Message]]:
                     or tp in active_partitions
                     or tp in self._buffered_partitions
                 ):
-                    highwater_mark = self.highwater(tp)
-                    self.app.monitor.track_tp_end_offset(tp, highwater_mark)
-                    # convert timestamp to seconds from int milliseconds.
-                    yield tp, to_message(tp, record)
+                    try:
+                        highwater_mark = self.highwater(tp)
+                        self.app.monitor.track_tp_end_offset(tp, highwater_mark)
+                        # convert timestamp to seconds from int milliseconds.
+                        yield tp, to_message(tp, record)
+                    except StopAsyncIteration:
+                        raise
+                    except Exception as exc:
+                        self.log.exception(
```

Comment:
same as above, this will cause all messages to be acknowledge and commit their offset even when the application fails to process it